### PR TITLE
install MetalLB as part of dependencies

### PR DESCRIPTION
**Pull request by dahorak**
Add possibility to install and configure MetalLB operator independently on ODF/hosted clusters deployment. This is required for Provider-Client pipeline where are all dependencies installed as part of separated stage.

### CODE REVIEW

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -611,6 +611,15 @@ def do_deploy_cnv(self):
         ):
             CNVInstaller().deploy_cnv()
 
+    def do_deploy_metallb(self):
+        """
+        Deploy MetalLB
+        """
+        if config.DEPLOYMENT.get("metallb_operator"):
+            from ocs_ci.deployment.metallb import MetalLBInstaller
```
Comment (vavuthu) [inline ocs_ci/deployment/deployment.py:619]: can we move it to import section or if there is any circular dependency issue, please add a comment

**Comment by dahorak** [comment]
Verification job: https://url.corp.redhat.com/42b231f

File: ocs_ci/deployment/deployment.py (hunk 1 above)
Comment (dahorak) [reply ocs_ci/deployment/deployment.py]: I'll check that, I've placed it there similarly as it is in the following method `do_deploy_hosted_clusters`,  but I'll check if there actually is some circular dependency issue and update accordingly.

Comment (dahorak) [reply ocs_ci/deployment/deployment.py]: Fixed by [33b3d46](https://github.com/red-hat-storage/ocs-ci/pull/10836/commits/33b3d4647eebb8d73d04f25b83c1a6b76c9a6b81)

**Review by vavuthu — APPROVED** [review]

**Pull request merged**